openwisp_firmware_upgrader/upgraders/openwrt.py

```diff
@@ -90,32 +91,37 @@ def _test_checksum(self, image):
                     'Identical checksum found in the filesystem, '
                     'upgrade not needed.'
                 )
+                message = _(
+                    'Firmware already upgraded previously. '
+                    'Identical checksum found in the filesystem, '
+                    'upgrade not needed.'
+                )
```

Comment:
This is not what I meant. Can you see the problem here? `message` variable is defined twice now. The suggestions were given just to give an idea what is expected. They were never meant to be used directly.

Reply:
Sorry, I missed that, I have committed it
I am still failing some tests and I am finding it hard to solve them.